IQSS/dataverse-frontend
# 187 - Files table download button

## What this PR does / why we need it:
This PR add the Download Button to the Files Table. (Only the UI with no backend integration yet)

## Which issue(s) this PR closes:

- Closes #187 

## Special notes for your reviewer:

## Suggestions on how to test this:
1. Install the necessary dependencies using `npm install`.
5. Build the design system with `cd packages/design-system && npm run build`.
6. Navigate back to the root directory with `cd ../../`.
7. Start Storybook with `npm run storybook`.
8. Visit [the Dataset in the Storybook](http://localhost:6006/?path=/story/pages-dataset--draft-with-all-dataset-permissions) to check that the Download button is being displayed in the Files Table, also check the behaviour about showing the No Selected Files Modal if no files are being selected

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
### Mockup of the requested element
![Captura de pantalla 2023-10-02 a las 11 11 10](https://github.com/IQSS/dataverse-frontend/assets/23359572/ae7bd489-fb6e-45e4-b78f-bd7853925b2b)

### For tabular data this is a dropdown
![Screen Shot 2023-10-05 at 2 49 22 PM](https://github.com/IQSS/dataverse-frontend/assets/2364928/4070a37d-f489-4af1-a2e0-da49976b3aea)

## Is there a release notes update needed for this change?:
Download button added to the Files Table UI

## Additional documentation:
